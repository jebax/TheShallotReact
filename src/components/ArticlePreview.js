import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ArticleImage from './ArticleImage'
import Headline from './Headline'

export default class ArticlePreview extends Component {
  render() {
    return(
      <div
        className='article-preview'
      >
        <ArticleImage
          imageUrl={this.props.imageUrl}
        />
        <Headline
          id={this.props.id}
          headline={this.props.headline}
          url={this.props.url}
        />
      </div>
    )
  }
}

ArticlePreview.propTypes = {
  id: PropTypes.number.isRequired,
  imageUrl: PropTypes.string.isRequired,
  headline: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
