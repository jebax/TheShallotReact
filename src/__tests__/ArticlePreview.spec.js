import React from 'react'
import { shallow } from 'enzyme'
import ArticlePreview from '../components/ArticlePreview'

describe('<ArticlePreview />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(
      <ArticlePreview
        id={0}
        imageUrl={'TestThumbnailUrl'}
        headline='TestHeadline'
        url='TestUrl'
      />
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a Headline', () => {
    expect(wrapper.find('Headline').length).toEqual(1)
  })

  it('renders an ArticleImage', () => {
    expect(wrapper.find('ArticleImage').length).toEqual(1)
  })

  it('renders an article preview wrapper', () => {
    expect(wrapper.find('.article-preview').length).toEqual(1)
  })
})
