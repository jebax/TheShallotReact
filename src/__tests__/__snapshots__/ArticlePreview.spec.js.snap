// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlePreview /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticlePreview
    headline="TestHeadline"
    id={0}
    imageUrl="TestThumbnailUrl"
    url="TestUrl"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArticleImage
          imageUrl="TestThumbnailUrl"
        />,
        <Headline
          headline="TestHeadline"
          id={0}
          url="TestUrl"
        />,
      ],
      "className": "article-preview",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "imageUrl": "TestThumbnailUrl",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "headline": "TestHeadline",
          "id": 0,
          "url": "TestUrl",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArticleImage
            imageUrl="TestThumbnailUrl"
          />,
          <Headline
            headline="TestHeadline"
            id={0}
            url="TestUrl"
          />,
        ],
        "className": "article-preview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "imageUrl": "TestThumbnailUrl",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "headline": "TestHeadline",
            "id": 0,
            "url": "TestUrl",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
